apps/server/src/modules/tool/uc/context-external-tool.uc.spec.ts

```diff
@@ -105,4 +105,50 @@ describe('ContextExternalToolUc', () => {
 			});
 		});
 	});
+
+	describe('deleteContextExternalTool is called', () => {
+		describe('when contextExternalTool is given and user has permission ', () => {
+			const setup = () => {
+				const userId: EntityId = 'userId';
+
+				const contextExternalTool: ContextExternalToolDO = contextExternalToolDOFactory.buildWithId();
+
+				const context = {
+					action: Action.write,
+					requiredPermissions: [Permission.CONTEXT_TOOL_ADMIN],
+				};
+
+				authorizationService.checkPermissionByReferences.mockResolvedValue(Promise.resolve());
```

Comment:
```suggestion
				authorizationService.checkPermissionByReferences.mockResolvedValue();
```
mockResolvedValue returns a resolved promise, see: https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvaluevalue